plugin/ochttp/server_test.go

```diff
@@ -273,3 +276,78 @@ func TestHandlerProxiesHijack_HTTP2(t *testing.T) {
 		t.Errorf("got = %q; want = %q", g, w)
 	}
 }
+
+func TestEnsureTrackingResponseWriterSetsStatusCode(t *testing.T) {
+	// Ensure that the trackingResponseWriter always sets the spanStatus on ending the span.
+	// Because we can only examine the Status after exporting, this test roundtrips a
+	// couple of requests and then later examines the exported spans.
+	// See Issue #700.
+	exporter := &spanExporter{cur: make(chan *trace.SpanData, 1)}
+	trace.RegisterExporter(exporter)
+	defer trace.UnregisterExporter(exporter)
+
+	tests := []struct {
+		res  *http.Response
+		want trace.Status
+	}{
+		{res: &http.Response{StatusCode: 200}, want: trace.Status{Code: 0, Message: `"OK"`}},
+		{res: &http.Response{StatusCode: 500}, want: trace.Status{Code: 2, Message: `"UNKNOWN"`}},
+		{res: &http.Response{StatusCode: 403}, want: trace.Status{Code: 7, Message: `"PERMISSION_DENIED"`}},
+		{res: &http.Response{StatusCode: 401}, want: trace.Status{Code: 16, Message: `"UNAUTHENTICATED"`}},
+		{res: &http.Response{StatusCode: 429}, want: trace.Status{Code: 8, Message: `"RESOURCE_EXHAUSTED"`}},
+	}
+
+	for i, tt := range tests {
+		t.Run(tt.want.Message, func(t *testing.T) {
+			span := trace.NewSpan("testing", nil, trace.StartOptions{Sampler: trace.AlwaysSample()})
+			ctx := trace.WithSpan(context.Background(), span)
+			prc, pwc := io.Pipe()
+			go func() {
+				pwc.Write([]byte("Foo"))
+				pwc.Close()
+			}()
+			inRes := tt.res
+			inRes.Body = prc
+			tr := &traceTransport{base: &testResponseTransport{res: inRes}}
+			req, err := http.NewRequest("POST", "https://example.org", bytes.NewReader([]byte("testing")))
+			if err != nil {
+				t.Fatalf("#%d: NewRequest error: %v", i, err)
```

Comment:
t.Fatalf("NewRequest error: %v", err)

Don't print i anymore. The messages will be outputted under the test case name.

Reply:
For shizzle, thanks.